doc/reference/libc/index.rst

```diff
@@ -0,0 +1,54 @@
+.. _libc_api:
+
+C standard library
+##################
+
+.. contents::
+    :local:
+    :depth: 2
+
+The `C standard library`_ is an integral part of any C program, and Zephyr
+provides two implementations for the application to choose from.
+
+The first one, named "minimal libc" is part of the Zephyr code base and provides
```

Comment:
Actually, I wonder if it'd be better just to get rid of the user guide and move all the info to the new reference section I've added.

Reply:
Sure, IMO, that would be less confusing.

Reply:
On second thought, keeping a "Standard C Library" doc under the user guide wouldn't necessarily be a bad thing since it probably is of interest to many people who've just started using Zephyr. Maybe we can keep the briefs in the existing doc and move all the details into this new file?

Reply:
Thanks for the input @stephanosio. I discussed this with @nashif and we agree that we need to start moving information from individual guides to the main reference section, so I will implement your first suggestion and add a redirect.

Reply:
is this going to be part of this PR?

Reply:
Yes, just added it now @nashif